packages/block-library/src/html/preview.js

```diff
@@ -23,7 +23,7 @@ const DEFAULT_STYLES = `
 
 export default function HTMLEditPreview( { content, isSelected } ) {
 	const settingStyles = useSelect( ( select ) => {
-		return select( blockEditorStore ).getSettings()?.styles;
+		return select( blockEditorStore ).getSettings()?.styles ?? [];
```

Comment:
It's better to define an `EMPTY_ARRAY` constant on the module level and use it here. Otherwise `useSelect` returns a new empty array on each call, which then defeats the memoization few lines below and leads to unnecessary rerenders.

This pattern is used very frequently in the codebase, can be found by grepping for `EMPTY_ARRAY` or `EMPTY_OBJECT`.

Reply:
@jsnajdr turns out we don't need to be that conservative in this selector. There are other instances where we're querying the same thing (`settings.styles`) and they're not checking if it exists.